pallets/auctions/src/types/english.rs

```diff
@@ -0,0 +1,169 @@
+//      ---_ ......._-_--.        ,adPPYba, 8b,dPPYba,    ,adPPYba,  88   ,d8
+//     (|\ /      / /| \  \       I8[    "" 88P'   `"8a  a8P_____88  88 ,a8"
+//     /  /     .'  -=-'   `.      `"Y8ba,  88       88  8PP"""""""  8888[
+//    /  /    .'             )    aa    ]8I 88       88  "8b,   ,aa  88`"Yba,
+//  _/  /   .'        _.)   /     `"YbbdP"' 88       88   `"Ybbd8"'  88   `Y8a
+//  / o   o        _.-' /  .'
+//  \          _.-'    / .'*|
+//  \______.-'//    .'.' \*|      This file is part of Basilisk-node.
+//   \|  \ | //   .'.' _ |*|      Built with <3 for decentralisation.
+//    `   \|//  .'.'_ _ _|*|
+//     .  .// .'.' | _ _ \*|      Copyright (C) 2021-2022  Intergalactic, Limited (GIB).
+//     \`-|\_/ /    \ _ _ \*\     SPDX-License-Identifier: Apache-2.0
+//      `/'\__/      \ _ _ \*\    Licensed under the Apache License, Version 2.0 (the "License");
+//     /^|            \ _ _ \*    you may not use this file except in compliance with the License.
+//    '  `             \ _ _ \    http://www.apache.org/licenses/LICENSE-2.0
+//     '  `             \ _ _ \
+
+use super::*;
+
+///
+/// Implementation of EnglishAuction
+///
+impl<T: Config> NftAuction<T::AccountId, T::AuctionId, BalanceOf<T>, Auction<T>, Bid<T>> for EnglishAuction<T> {
+	///
+	/// Creates an English Auction
+	///
+	#[require_transactional]
```

Comment:
aren't all fns transactional by default?

Reply:
@apopiak can you advise on this? I added the require_transactional upon your request in the original PR.

Reply:
extinsics are transactional by default, yes, but I think it makes sense to still tag functions that require this as the function could be called from an extrinsic that opts out of transactionality or could be called from `on_initialize` which is not transactional by default IIRC.

Reply:
makes sense, esp. with helper fns